### PR TITLE
417 - Update alert for round apply

### LINKED ISSUE #417: Success banner after successful round apply
**User Story**
As a project owner, 
I want to get a success confirmation after I've applied to a round
so that I know my application went through 

**Acceptance Criteria**
GIVEN that I submitted my application to a round
WHEN I am redirected to the Project Details page
THEN I see a success banner 

The final copy for success banner: 
"Thank you for applying to [Program Name] [Round Name]! 
Your application has been received, and the [Program Name] team will review and reach out with next steps."

**Tech Details**: 
* Related to: https://github.com/gitcoinco/grant-hub/pull/387

**Designs**
Figma: https://www.figma.com/file/obBzCCL8KwoTlVBb5g3ccN/Grant-Hub---Wireframes?node-id=2403%3A14112

Screen Shot 2022-09-21 at 5 41 18 PM

### PR DESCRIPTION
**Pull request by codenamejason**

### CODE REVIEW

**Comment by codenamejason** [comment]
Screen Shot 2022-10-04 at 4 28 50 PM

**Review by ghostffcode — APPROVED** [review]
LGTM!

**Comment by michellema1208** [comment]
@codenamejason this is looking so, so good! Great job so far :)

I don't have any product functionality issues to call out, but I did notice that the font style + sizing looked a bit different from what Will had in designs: 

**From my test application**: 
Screen Shot 2022-10-05 at 3 56 31 PM

**From Designs**: 
Screen Shot 2022-10-05 at 3 58 19 PM

I'm still a bit of a Figma Novice, but it looks like the font style Will is using is Libre Franklin, and font size is 14px - more details on the right hand side under Typography. I think the only other thing is that the message is **bolded**, when it should just be normal text (not bolded). If @willsputra is still online, I'll let him test / confirm as well! 

So to recap, some style tasks would be: 

- [ ] Unbold the copy in the messages (both the header and the body text) 
- [ ] Update copy to match the font size & style Will is using (see the Typography section in Figma / in my screenshot)

**Comment by gravityblast** [comment]
can we remove the check.svg file and use the heroicons in case we need it?

File: client/src/components/base/Alert.tsx [hunk 1]
```diff
@@ -1,8 +1,9 @@
 import { useEffect } from "react";
 import { useDispatch } from "react-redux";
-import Check from "../icons/Check";
-import { Alert } from "../../types/alert";
+import { CheckCircleIcon } from "@heroicons/react/24/solid";
 import { removeAlertDelayed } from "../../actions/ui";
+import { Alert } from "../../types/alert";
+// import Check from "../icons/Check";
```
Comment (gravityblast) [inline client/src/components/base/Alert.tsx]: maybe we forgot to remove this comment

File: client/src/components/base/Alert.tsx [hunk 2]
```diff
@@ -20,21 +21,20 @@ export default function AlertComponent({ alert }: AlertProps) {
...
+        <div className="block">{alert.title}</div>
+        <div className="block text-black">{alert.body}</div>
+      </div>
     </div>
   );
 }
 
 export function AlertContainer({ alerts }: AlertContainerProps) {
-  if (alerts.length === 0) {
+  if (alerts?.length === 0) {
```
Comment (gravityblast) [inline client/src/components/base/Alert.tsx]: we can remove the `?` since `alerts` is not optional

File: client/src/components/base/Preview.tsx [hunk 3]
```diff
@@ -56,7 +56,7 @@ export default function Preview({
         navigate(slugs.grants);
         localResetStatus();
         dispatch(
-          addAlert("success", "Your project has been saved successfully!")
+          addAlert("success", "Your project has been saved successfully!", "")
```
Comment (gravityblast) [inline client/src/components/base/Preview.tsx]: I would actually pass `undefined` instead of empty string so it feels better when we check if there's a body or not

File: client/src/components/grants/inputs.tsx [hunk 4]
```diff
@@ -72,21 +73,7 @@ export function TextInputAddress({
...
-            <path
-              fillRule="evenodd"
-              d="M2.25 12c0-5.385 4.365-9.75 9.75-9.75s9.75 4.365 9.75 9.75-4.365 9.75-9.75
-              9.75S2.25 17.385 2.25 12zm8.706-1.442c1.146-.573 2.437.463 2.126 1.706l-.709
-              2.836.042-.02a.75.75 0 01.67 1.34l-.04.022c-1.147.573-2.438-.463-2.127-1.706l.71-2.836-.042.02a.75.75
-              0 11-.671-1.34l.041-.022zM12 9a.75.75 0 100-1.5.75.75 0 000 1.5z"
-              clipRule="evenodd"
-            />
-          </svg>
+          <InformationCircleIcon className="w-6 h-6" color="gray" />
```
Comment (gravityblast) [inline client/src/components/grants/inputs.tsx:77]: that's great we should slowly do this for all the hardcoded icons we have

File: client/src/components/rounds/Apply.tsx [hunk 5]
```diff
@@ -74,6 +92,7 @@ function Apply() {
   }, [roundId]);
 
   if (props.roundStatus === RoundStatus.Error) {
+    dispatch(addAlert("error", applicationErrorTitle, applicationErrorBody));
```
Comment (gravityblast) [inline client/src/components/rounds/Apply.tsx]: is this defined in any design? I like the idea of having the alert, but do we have an empty html body or do we redirect somewhere?

File: client/src/components/rounds/Apply.tsx [hunk 6]
```diff
@@ -91,9 +110,11 @@ function Apply() {
     );
   }
 
-  // todo: navigating to early...
   if (props.applicationStatus === ApplicationStatus.Sent) {
     if (props.applicationState) {
+      dispatch(
+        addAlert("success", applicationSuccessTitle, applicationSuccessBody)
```
Comment (gravityblast) [inline client/src/components/rounds/Apply.tsx:140]: 🚀

File: client/src/types/alert.ts [hunk 7]
```diff
@@ -1,16 +1,18 @@
...
-  message: string;
+  title: string;
+  body: string;
 };
 
 let nextAlertID = 0;
 
-export const newAlert = (type: AlertTypes, message: string) => ({
+export const newAlert = (type: AlertTypes, title: string, body: string) => ({
   // eslint-disable-next-line
```
Comment (gravityblast) [inline client/src/types/alert.ts:13]: I actually hate eslint lol :)

**Review by gravityblast — APPROVED** [review]
LGTM! 🚀 I just added a couple of comments

File: client/src/actions/ui.ts [hunk 8]
```diff
@@ -15,9 +15,13 @@ export interface UIAlertRemovedAction {
 
 export type UIActions = UIAlertAddedAction | UIAlertRemovedAction;
 
-export const addAlert = (type: AlertTypes, message: string): UIActions => ({
+export const addAlert = (
+  type: AlertTypes,
+  title: string,
+  body: string
```
Comment (gravityblast) [inline client/src/actions/ui.ts]: this could be `string | undefined` if we want

**Comment by willsputra** [comment]
image

this is what I see. agree with @michellema1208's changes, otherwise it's looking great!
> Unbold the copy in the messages (both the header and the body text)
 Update copy to match the font size & style Will is using (see the Typography section in Figma / in my screenshot)

**Comment by codenamejason** [comment]
This is 14px normal weight(400) Libre Franklin

Does this look right? I cant tell by looking at it... my eyes are doing tricks on me now 👀

Screen Shot 2022-10-05 at 4 40 33 PM

File: client/src/components/rounds/Apply.tsx (hunk 5 above)
Comment (codenamejason) [reply client/src/components/rounds/Apply.tsx]: I was thinking about that earlier... I was just using a generic message in the alert and html body for now and does not redirect. @gravityblast

**Comment by michellema1208** [comment]
@codenamejason yes your screenshot looks perfect! With that change, LGTM [🚀](https://emojipedia.org/rocket/)

File: client/src/components/rounds/Apply.tsx (hunk 5 above)
Comment (codenamejason) [reply client/src/components/rounds/Apply.tsx]: also had it in the wrong spot 🤦‍♂️
